src/components/metadata/metadata.js

```diff
@@ -85,7 +85,7 @@ const MetaData = ({
   };
 
   const shortenDatasetType = (string) => {
-    return string?.split('.').pop();
+    return string ? string.split('.').pop() : 'MemoryDataSet';
```

Comment:
I think this fixes the issue, but I'm not sure whether it's expected that `undefined` gets passed in here?

Reply:
I think this is not expected (at least by me anyway). I'd expect the dataset type here to be `kedro.io.core.MemoryDataSet`.

Reply:
For some reason the responses for memory datasets are empty. [See here](https://demo.kedro.org/api/nodes/c08c7708), compared with the response for a [CSV dataset](https://demo.kedro.org/api/nodes/aed46479).

Given that, this does fix the issue. All other datasets return what we expect from the endpoint.

Reply:
> For some reason the responses for memory datasets are empty. [See here](https://demo.kedro.org/api/nodes/c08c7708), compared with the response for a [CSV dataset](https://demo.kedro.org/api/nodes/aed46479).

I think this may be the case only for `MemoryDataSet`s not explicitly listed in the catalog. Let me test using the API (thanks for showing that!) as soon as my other issue resolves itself:


https://user-images.githubusercontent.com/14007150/193087503-14f548b2-4363-4c63-b297-3f5d8ab698d6.mov

Reply:
> For some reason the responses for memory datasets are empty. [See here](https://demo.kedro.org/api/nodes/c08c7708), compared with the response for a [CSV dataset](https://demo.kedro.org/api/nodes/aed46479).

I think this may be the case only for `MemoryDataSet`s not explicitly listed in the catalog. Let me test using the API (thanks for showing that!) as soon as my other issue resolves itself:

![193087503-14f548b2-4363-4c63-b297-3f5d8ab698d6](https://user-images.githubusercontent.com/14007150/193088273-6dd9ea16-2331-40e5-a33b-d63b6e8b6d80.gif)

Reply:
@tynandebold Just to sanity-check since I'm new to Viz...

If you explicitly add a catalog entry for a `MemoryDataSet`, does the `/api/nodes` call also give you and `Internal Server Error`? Maybe some `pydantic` validation error?

Reply:
I'm afraid I don't know the answer to this one.

Reply:
This is also where my knowledge of Kedro starts to break down. I tried to add a catalog entry for a `MemoryDataSet` but got an error:

```
DataSet 'train_evaluation.random_forest.experiment_params' must only contain arguments valid for the constructor of 'kedro.io.memory_dataset.MemoryDataSet'.
```

If you can instruct me here I can sanity check for you on my end.

Reply:
You can add a `MemoryDataSet` explicitly like:

```yaml
train_X:
  type: MemoryDataSet
```

I would probably do it for one of the datasets that isn't explicitly registered (e.g. `train_X` and every other dataset other than the input data in `pandas-iris`), but not for parameters. The only other thing you could specify in the catalog entry is `copy_mode`, but that's not necessary.

:)

Reply:
Nice thank you. I get a 500 error if I do that:

```
GET http://localhost:4141/api/nodes/cae2d1c7 500 (Internal Server Error)
```

Here's the server error:

```
2022-09-30 11:14:49,229 - uvicorn.access - INFO - None:0 - "GET /api/nodes/cae2d1c7 HTTP/1.1" 500
2022-09-30 11:14:49,229 - uvicorn.error - ERROR - Exception in ASGI application
Traceback (most recent call last):
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/uvicorn/protocols/http/httptools_impl.py", line 372, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/uvicorn/middleware/message_logger.py", line 82, in __call__
    raise exc from None
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/uvicorn/middleware/message_logger.py", line 78, in __call__
    await self.app(scope, inner_receive, inner_send)
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/fastapi/applications.py", line 199, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/starlette/routing.py", line 580, in __call__
    await route.handle(scope, receive, send)
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/starlette/routing.py", line 241, in handle
    await self.app(scope, receive, send)
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/starlette/routing.py", line 52, in app
    response = await func(request)
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/fastapi/routing.py", line 224, in app
    response_data = await serialize_response(
  File "/Users/tynandebold/opt/anaconda3/lib/python3.8/site-packages/fastapi/routing.py", line 127, in serialize_response
    raise ValidationError(errors, field.type_)
pydantic.error_wrappers.ValidationError: <unprintable ValidationError object>
```

Reply:
@tynandebold Thanks for verifying! I thought I must be doing something wrong earlier in my setup since I was getting such errors, but seems this is an actual issue, so I will investigate. :D